File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
function findMaxAverage(nums, k) {
    let currentSum = 0;

    // sum of first k elements
    for (let i = 0; i < k; i++) {
        currentSum += nums[i];
    }

    let maxSum = currentSum;

    // sliding window
    for (let i = k; i < nums.length; i++) {
        currentSum -= nums[i - k];
        currentSum += nums[i];

        maxSum = Math.max(maxSum, currentSum);
    }

    return maxSum / k;
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: list[int], k: int) -> float:
        current_sum = sum(nums[:k])
        max_sum = current_sum

        for i in range(k, len(nums)):
            current_sum -= nums[i - k]
            current_sum += nums[i]

            max_sum = max(max_sum, current_sum)

        return max_sum / k
